### PR TITLE
SER-1009 | make renaming username with semicolon invalid

### PR DESCRIPTION
**Pull request by idradm**
This reverts commit cf3c954ab6868c5be1317e5dabafa7eef1428bba. 

Adds colon as invalid username character for username rename of creation.
https://wikia-inc.atlassian.net/browse/SER-1009

### CODE REVIEW

File: includes/User.php [hunk 1]
```diff
@@ -680,7 +680,6 @@ public static function isValidUserName( $name ) {
 		if ( $name == ''
```
Comment (WikiaTech) [inline includes/User.php:680]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 115 issues
* ![MAJOR][MAJOR] 66 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 42 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [UserSignupTest.php#L107](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L107): This function "signupDataProvider" has 199 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L434](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L434): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L478](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L478): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L488](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L488): This function has 14 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L540](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L540): This function "sendConfirmationEmailDataProvider" has 280 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L619](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L619): Remove this unused "$mockMessagesMap101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L624](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L624): Remove this unused "$expMsg101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L632](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L632): Remove this unused "$mockMessagesMap102" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L637](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L637): Remove this unused "$expMsg102" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [User.php](https://github.com/Wikia/app/blob/f2ade3892947143df6f87c769907a252049d4a3e/includes/User.php): File "User.php" has 4702 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 212 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 104 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 98 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1048](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/includes/DefaultSettings.php#L1048): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1184](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/includes/DefaultSettings.php#L1184): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L5522](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/includes/DefaultSettings.php#L5522): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L107](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/extensions/wikia/UserLogin/tests/UserSignupTest.php#L107): This function "signupDataProvider" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L440](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/extensions/wikia/UserLogin/tests/UserSignupTest.php#L440): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L484](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/extensions/wikia/UserLogin/tests/UserSignupTest.php#L484): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L494](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/extensions/wikia/UserLogin/tests/UserSignupTest.php#L494): This function has 14 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L546](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/extensions/wikia/UserLogin/tests/UserSignupTest.php#L546): This function "sendConfirmationEmailDataProvider" has 280 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L625](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/extensions/wikia/UserLogin/tests/UserSignupTest.php#L625): Remove this unused "$mockMessagesMap101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L630](https://github.com/Wikia/app/blob/2801b0cfbc946b61ab87bcf6848642f2d04a64f4/extensions/wikia/UserLogin/tests/UserSignupTest.php#L630): Remove this unused "$expMsg101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by sqreek — APPROVED** [review]

File: extensions/wikia/UserLogin/tests/UserSignupTest.php [hunk 2]
```diff
@@ -268,9 +268,9 @@ public function signupDataProvider() {
 				'password' => '123',
 			);
 
-			$reqParams18 = array(
-				'userloginext01' => self::TEST_USERNAME . ':test' ,
-			);
+			$reqParams18 = [
```
Comment (sqreek) [inline extensions/wikia/UserLogin/tests/UserSignupTest.php]: minor:
If we change array to `[]` change it in whole file, or don't change it and keep `file` convention

**Comment by WikiaTech** [comment]
SonarQube analysis reported 212 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 104 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 98 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1048](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/includes/DefaultSettings.php#L1048): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1184](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/includes/DefaultSettings.php#L1184): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L5522](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/includes/DefaultSettings.php#L5522): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L107](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/extensions/wikia/UserLogin/tests/UserSignupTest.php#L107): This function "signupDataProvider" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L440](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/extensions/wikia/UserLogin/tests/UserSignupTest.php#L440): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L484](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/extensions/wikia/UserLogin/tests/UserSignupTest.php#L484): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L494](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/extensions/wikia/UserLogin/tests/UserSignupTest.php#L494): This function has 14 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L546](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/extensions/wikia/UserLogin/tests/UserSignupTest.php#L546): This function "sendConfirmationEmailDataProvider" has 280 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L625](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/extensions/wikia/UserLogin/tests/UserSignupTest.php#L625): Remove this unused "$mockMessagesMap101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L630](https://github.com/Wikia/app/blob/9ec8c3e8e7dc915ad113c28c341405144920363c/extensions/wikia/UserLogin/tests/UserSignupTest.php#L630): Remove this unused "$expMsg101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 212 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 104 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 98 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1048](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/includes/DefaultSettings.php#L1048): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L1184](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/includes/DefaultSettings.php#L1184): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL][CRITICAL] [DefaultSettings.php#L5522](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/includes/DefaultSettings.php#L5522): Remove this hard-coded password. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L107](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/extensions/wikia/UserLogin/tests/UserSignupTest.php#L107): This function "signupDataProvider" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L440](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/extensions/wikia/UserLogin/tests/UserSignupTest.php#L440): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L484](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/extensions/wikia/UserLogin/tests/UserSignupTest.php#L484): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L494](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/extensions/wikia/UserLogin/tests/UserSignupTest.php#L494): This function has 14 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L546](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/extensions/wikia/UserLogin/tests/UserSignupTest.php#L546): This function "sendConfirmationEmailDataProvider" has 280 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L625](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/extensions/wikia/UserLogin/tests/UserSignupTest.php#L625): Remove this unused "$mockMessagesMap101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L630](https://github.com/Wikia/app/blob/3bb4347819db32a7513e24c986e249bcf03f4cbb/extensions/wikia/UserLogin/tests/UserSignupTest.php#L630): Remove this unused "$expMsg101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**